nilearn/_utils/glm.py

```diff
@@ -145,7 +145,11 @@ def z_score(pvalue):
     """ Return the z-score corresponding to a given p-value.
     """
     pvalue = np.minimum(np.maximum(pvalue, 1.e-300), 1. - 1.e-16)
-    return norm.isf(pvalue)
+    if np.any(pvalue > .5):
+        zscore = - norm.isf(1 - pvalue)
+    else:
+        zscore = norm.isf(pvalue)
+    return zscore
```

Comment:
Actually, this function does not does not change anything for me. A way to soleve this is to have a test that fails with the previous versions and passes with the new one.

Reply:
The problem as reported by the author of the issue is reproducible in the latest version of nilearn, i.e. in vs. "0.7.0a". But in development, it is not possible to reproduce it exactly in the same way since we cannot download spm auditory data as pointed out by @jeromedockes and @KamalakerDadi. We need to produce fake data and be able to get z values > 7 in order to start seeing differences between the max value of a contrast and the min value of its reverse. Shall I still proceed with the development of the test function w/ fake data? Otherwise, what do you suggest?